tests/foreman/ui/test_organization.py

```diff
@@ -16,16 +16,56 @@
 
 :Upstream: No
 """
+import logging
+
 import pytest
 from fauxfactory import gen_string
 from nailgun import entities
 
 from robottelo.config import settings
 from robottelo.constants import DEFAULT_ORG
+from robottelo.constants import DISTRO_RHEL7
 from robottelo.constants import INSTALL_MEDIUM_URL
 from robottelo.constants import LIBVIRT_RESOURCE_URL
+from robottelo.constants import VALID_GPG_KEY_FILE
+from robottelo.helpers import read_data_file
 from robottelo.manifests import original_manifest
 from robottelo.manifests import upload_manifest_locked
+from robottelo.products import RepositoryCollection
+from robottelo.products import RHELAnsibleEngineRepository
+from robottelo.products import SatelliteToolsRepository
+from robottelo.products import YumRepository
+
+CUSTOM_REPO_URL = settings.repos.yum_6.url
+CUSTOM_REPO_ERRATA_ID = settings.repos.yum_6.errata[2]
+logger = logging.getLogger('robottelo')
+
+
+@pytest.fixture(scope='module')
+def module_repos_col(module_org, module_lce, default_sat, request):
+    upload_manifest_locked(org_id=module_org.id)
+    repos_collection = RepositoryCollection(
+        distro=DISTRO_RHEL7,
+        repositories=[
+            SatelliteToolsRepository(),
+            # As Satellite Tools may be added as custom repo and to have a "Fully entitled" host,
+            # force the host to consume an RH product with adding a cdn repo.
+            RHELAnsibleEngineRepository(cdn=True),
+            YumRepository(url=CUSTOM_REPO_URL),
+        ],
+    )
+    repos_collection.setup_content(module_org.id, module_lce.id)
+
+    @request.addfinalizer
```

Comment:
This pattern is quite redundant within a fixture function.

Just move the try/except block to after a `yield` where currently the function `return`s

https://docs.pytest.org/en/6.2.x/fixture.html#yield-fixtures-recommended

Reply:
Updated as per comments.